test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/irrule/constraint/raw/RawCountsConstraint.java

```diff
@@ -52,9 +56,42 @@ public CompilePhase defaultCompilePhase() {
         return rawIRNode.defaultCompilePhase();
     }
 
+    private boolean expectMaxSizeForVectorNode() {
+        switch (comparison.getComparator()) {
+            case "<" -> {
+                TestFormat.checkNoReport(comparison.getGivenValue() > 1, "Node count comparison \"<" +
```

Comment:
What if `comparison.getGivenValue()` is negative?

Reply:
A rule like that would always fail.

Reply:
Will add more checks to give more precise messages.